src/storage.ts

```diff
@@ -55,6 +55,11 @@ export interface StorageOptions extends GoogleAuthOptions {
    * Defaults to `www.googleapis.com`.
    */
   apiEndpoint?: string;
+  /**
+   * The scheme of the service used to make requests.
+   * Defaults to `https`.
+   */
+  apiScheme?: string;
```

Comment:
So this is an issue.  We don't specifically expose this as a property in any other library, and I'd be worried to have a snowflake here.  

@broady what are your thoughts?

Reply:
Aside - if we *do* decide to do this, we would also need to update the jsdoc annotation

Reply:
In other libraries - such as Go and Java - the host includes the scheme. In contrast, in Node, the host is everything after scheme up to port. Would you prefer me to make it conform in that way - by having scheme be part of `apiHost?`?

Reply:
Eugh, gross.  Here's my thoughts:
- We should probably have the `apiEndpoint` accept an optional schema prefix
- We should have a utility function in `nodejs-common` that is responsible for parsing the apiEndpoint, determining if it has a schema (or port), then assembling the final thing
-  Roll this out more broadly to other nodejs libraries so it's consistent.

That having all be said - I would rather not do that.  If the scenario here is a local emulator - can we rely on an env var (which makes more sense IMO) to help us do . the right thing?  I think that's what we're doing in PubSub. 

https://github.com/googleapis/nodejs-pubsub/blob/master/src/pubsub.ts#L524

Reply:
Happy to go the env approach! We do the same in Go with Pub/Sub, Storage, Spanner, Firestore, and BigQuery IIRC. (`<SERVICE>_EMULATOR_HOST`) Will make the adjustment

Reply:
any reason we can't just grab `URL.parse(apiEndpoint)` and fetch the scheme off that.

Reply:
I personally prefer the env var: it's simpler for the benchwrapper (don't need --emulatorHost), it's less complicated than string parsing, and conforms with how Java and Go work.

Reply:
Done